value/eval.go

```diff
@@ -473,3 +469,17 @@ func isTrue(fnName string, v Value) bool {
 		return false
 	}
 }
+
+// fixScanOp returns the appropriate op for scan operations, according
+// to it, the iteration variable.
+func fixScanOp(op string, it int) string {
+	if it%2 == 0 {
+		switch op {
```

Comment:
There are a lot of other operators available. While the reduction in complexity is admirable, the fix should be universal or at least only apply when it is appropriate.

Or else I misunderstand, which is possible.

Reply:
Right, I missed that.
I now also see in [Wikipedia](https://en.wikipedia.org/wiki/Associative_property) that operations are not associative for floating point numbers.
I guess I can drop this PR then.